src/_guides/openapi/advanced-ref-usage.md

```diff
@@ -0,0 +1,335 @@
+
+Other complications can appear depending on which tools you're using. Some tools do not support URLs in `$ref`, either for security concerns or because the tool maintainers never got around to it. You need to programmatically replace all the $ref's with URLs to be local refs, and whilst there are tools which can "bundle" your API descriptions up for you, it's another bit of work, and adds another copy of the API description document to keep track of and keep updated.
+
+Finally there's authentication. Some people have their API descriptions in a private Git repositories and cannot access it with `https://raw.githubusercontent.com/org/repo/main/content/schemas/foo.json` because it would need some sort of access token and how's that going to work? Making a GitHub Action / Continuous Integration step that deploys the API descriptions or schemas to a public S3 bucket or other public static site is probably the best thing to do there.
+
+Others hide their OpenAPI and AsyncAPI by choice for security reasons, but that's never made much sense because Stripe, PayPal, Box, GitHub, and plenty of other massive API companies have their API descriptions out in public and nobody has hacked them. APIs should be protected with firewalls and API keys, but OpenAPI and AsyncAPI information can be plastered all over the place. Another vote for the public static site.
+
+There is an argument for making public APIs public and keeping internal API's private, and some hosted API documentation tools can help with that, or you can host internal API docs on a different static site that's only available on the company network. Either way you'll need to keep your public APIs public, and keep your shared components public, then hide the internal APIs that reference those. That gives you the best of both worlds.
+
+## Propagating Changes
+
+Using tools like Bump.sh you get all the benefits of a tool that understands $ref, but without any of the hassle of needing to bundle documents up. 
+
+Like any tool which uses a build step, this has the pro and the con of meaning that documentation is built at a certain point in time. Changes that happen to the $ref'ed resources - whether they're in another repository, or being pulled in via URL - will take some time to appear in your API.
+
+For example, if the Widget API is using a shared Company schema via `$ref: https://widgets.com/schema/company.json`, and company decides to add VAT number as a property, your Widget API documentation is not going to mention that property until your next build.
+
+Is that a good thing or a bad thing? It can be both depending on the scenario, but having changes appear in your API without your knowledge is probably not ideal.
```

Comment:
:+1: completely agree with this statement, maybe we can also mention the fact that the code of the API will probably need a change/build too anyways.